src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -26,24 +26,6 @@ public class QnAService {
     @Transactional
     public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
         Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
+        deleteHistoryService.saveAll(question.delete(loginUser));
```

Comment:
```suggestion
        question.delete(loginUser);
        deleteHistoryService.saveAll(question.toDeleteHistories());
```

위와 같이 삭제 메시지와 삭제 히스토리 생성하는 부분을 분리하는 것과 위와 같이 하나의 메서드를 처리하는 것 중 어느 것이 더 나을까?
각각의 장단점은 무엇일까?

Reply:
1. 분리 시
 장점 : 
  1) 코드 가독성이 증가합니다. 각각의 메소드가 어떤 역할을 하는 지 다른 개발자들이 보았을 떄 보기 쉬울 것 같습니다. 
  2) 분리를 통한 테스트가 용이합니다. 각각의 메소드에 대한 테스트코드를 명확하게 작성할 수 있습니다. 
  3) 각각의 메소드에 대한 재사용성이 증가합니다.
  
2. 분리 X
분리하지 않는 것에 대한 특별한 장점을 느끼지 못했습니다.
그나마 코드가 간결해진다는 정도?로 생각하면 될까요?